at_client/src/main/java/org/atsign/client/util/RegisterUtil.java

```diff
@@ -100,42 +109,86 @@ public Boolean registerAtsign(String email, AtSign atsign, String registrarUrl,
      * if OTP is valid.
      * Returns the CRAM secret of the atsign which is registered.
      * 
-     * @param email
-     * @param atsign
-     * @param otp
-     * @param registrarUrl
-     * @return
-     * @throws IOException
-     * @throws AtException
+     * @param email        - email initially provided to register this email
+     * @param atsign       - atsign that is being registered
+     * @param otp          - one-time-password received on provided email for
+     *                     provided atsign
+     * @param registrarUrl - URL of the atsign registrar API
+     * @param apiKey       - API key to authenticate connection to atsign registrar
+     *                     API
+     * @param confirmation - Mandatory parameter for validateOTP API call. First
+     *                     request to be sent with confirmation as false, in this
+     *                     case API will return cram key if the user is new
+     *                     otherwise will return list of already existing atsigns.
+     *                     If the user already has existing atsigns user will have
+     *                     to select a listed atsign and place a second call to the
+     *                     same API endpoint with confirmation set to true with
+     *                     previously received OTP.
+     * @return Case 1("verified") - the API has registered the atsign to
+     *         provided email and cramkey present in HTTP_RESPONSE Body.
+     *         Case 2("follow-up"): User already has existing atsigns and new atsign
+     *         registered usccessfully. To receive cramkey follow-up with API with
+     *         one
+     *         of exsting listed atsigns with confirmation set to true.
+     *         Case 3("retry"): Incorrect OTP send request again with correct OTP.
+     * @throws IOException if anything goes wrong while handling I/O streams from
+     *                     HttpsURLConnection.
+     * @throws AtException Case 1: If user has exhausted 10 free atsign quota
+     *                     Case 2: If API response is anything other than
+     *                     HTTP_OK/Status_200.
      */
-    public String validateOtp(String email, AtSign atsign, String otp, String registrarUrl, String apiKey)
+    public String validateOtp(String email, AtSign atsign, String otp, String registrarUrl, String apiKey,
+            boolean ... confirmation)
             throws IOException, AtException {
+        //setting default confirmation to true in case it's not provided
+        Boolean defaultConfirmation = (confirmation.length == 0) ? true : confirmation[0];
```

Comment:
This is not what I was looking for. I asked for the confirmation parameter to be made optional, I was thinking as a named optional parameter e.g. 
`public String validateOtp(String email, AtSign atsign, String otp, String registrarUrl, String apiKey, {boolean confirmation = true})`

Reply:
yes, as far as StackOverflow goes this is one of the limited ways we can do optional parameters

Reply:
Ignore my previous comments

To preserve backwards compatibility
1. Add new method `public String validateOtp(String email, AtSign atsign, String otp, String registrarUrl, String apiKey, boolean confirmation)`
2. Keep existing method `public String validateOtp(String email, AtSign atsign, String otp, String registrarUrl, String apiKey)` with implementation `{validateOtp(email, atsign, otp, registrarUrl, apiKey, true); }` assuming that the default value for 'confirmation' is true